syntax = "proto3";

import "google/protobuf/struct.proto";
import "anfrage.proto";

message SupplierToRequest {
  string id = 1;
  google.protobuf.Struct contacts = 2;
  google.protobuf.Struct metadata = 3;
}

message CampaignSchedule {
  string start_at = 1;
  string end_at = 2;
}

message CampaignState {
  google.protobuf.Struct states = 1;
}

// A message to be body of purchase_campaing.verified.run service's request
message PurchaseCampaignVerified {
  int64 id = 1;
  pce.bo.Anfrage anfrage = 2;
  repeated SupplierToRequest suppliers_to_request = 3;
  CampaignSchedule schedule = 4;
  int64 state = 5;
//  google.protobuf.Struct metadata = 6;
}

message PurchaseCampaingUnverified {
  string id = 1;
  string free_text_anfrage = 2;
  google.protobuf.Struct metadata = 6;
}

message VoidRequest {}

service PurchaseCampaing {
    rpc CreatePurchaseCampanies(PurchaseCampaignVerified) returns (PurchaseCampaignVerified) {};
    rpc GetPurchaseCampanies(VoidRequest) returns (PurchaseCampaignVerified) {};
}